osquery/tables/system/posix/ssl_cert.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <openssl/asn1.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+static void fillRow(Row& r, X509* cert) {
+#define SHORT_STR 256
+  char temp[SHORT_STR + 1];
+
+  // set certificate subject information
+  auto subject_name = X509_get_subject_name(cert);
+  auto ret =
+      X509_NAME_get_text_by_NID(subject_name, NID_commonName, temp, SHORT_STR);
```

Comment:
Similarly, once you've swapped out `temp` with a `vector`, you can get it's size here with `temp.size()`